### PR TITLE
Email

### PR DESCRIPTION
**Pull request by btimby**
I need a way to create a `.msg` file to test that backend.

### CODE REVIEW

File: tests.py [hunk 1]
```diff
@@ -261,6 +261,14 @@ class PsTestCase(BaseTestCase, PathAndFileTests):
...
 
 
+class EmlTestCase(BaseTestCase, PathAndFileTests):
+    ext = "eml"
+
+
+class MboxTestCase(BaseTestCase, PathAndFileTests):
+    ext = "mbox"
+
+
```
Comment (giampaolo) [inline tests.py]: No `.msg`?

File: setup.py [hunk 2]
```diff
@@ -1,20 +1,26 @@
...
+
 with open('requirements.txt') as f:
-    REQUIRED = f.read().splitlines()
+    for line in f.readlines():
+        if 'http://' in line or 'https://' in line:
+            REQUIRED_URL.append(line)
 
-REQUIRED = [r for r in REQUIRED if not r.startswith('git')]
+        else:
+            REQUIRED.append(line)
```
Comment (giampaolo) [inline setup.py]: nice

File: fulltext/backends/__eml.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+def _get_file(f, **kwargs):
+    text, f = StringIO(), codecs.getreader('utf8')(f, errors='ignore')
+    m = message_from_file(f)
+
+    for part in m.walk():
+        if part.get_content_type().startswith('text/plain'):
+            text.write(part.get_payload())
+
+    return text.getvalue()
```
Comment (giampaolo) [inline fulltext/backends/__eml.py]: file `f` should be closed

File: fulltext/backends/__eml.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
+from email import message_from_file
+
+import codecs
+
+from six import StringIO
+
+
+def _get_file(f, **kwargs):
+    text, f = StringIO(), codecs.getreader('utf8')(f, errors='ignore')
```
Comment (giampaolo) [inline fulltext/backends/__eml.py]: nice

File: tests.py (hunk 1 above)
Comment (btimby) [reply tests.py]: I don't have a way to create a test file just yet. I should soon, and I will add it and merge.

File: fulltext/backends/__eml.py (hunk 3 above)
Comment (btimby) [reply fulltext/backends/__eml.py]: `f` belongs to caller. I am wrapping it in a `StreamReader`, but if I close _that_ it will close `f` too.

Comment (giampaolo) [reply fulltext/backends/__eml.py]: oh you're right

**Pull request merged**